portfolio/src/main/webapp/script.js

```diff
@@ -109,22 +178,164 @@ function retrieveId(string, source){
 }
 
 /**
-  * Runs the search by getting the source HTML, getting the appropriate tag, and scrolling towards that id
-  */
+ * Get's all id's from a given tag array
+ */
+function getIds(array, source){
+    var retArr = [];
+    for (var i = 0; i < array.length; i++){
+        var retriever = retrieveId(array[i], source);
+        if (retriever === "nothing"){
+            continue;
+        } else {
+            retArr.push(retriever);
+        }
+    }
+
+    return retArr;
+}
+
+/**
+ * Counts the number of duplicate elements in an array
+ * NOT USED
```

Comment:
will this be used in a future PR? consider removing for now and adding back in when it's needed